cdap-watchdog/src/main/java/co/cask/cdap/logging/pipeline/kafka/KafkaLogProcessorPipeline.java

```diff
@@ -288,147 +289,82 @@ private boolean processMessages(String topic, int partition,
-        // Since the offset is not updated, the same set of messages will be fetched again in next iteration.
-        int eventsAppended = appendEvents(System.currentTimeMillis(), true);
-        if (eventsAppended <= 0) {
-          break;
-        }
-        unSyncedEvents += eventsAppended;
-      }
-
-      try {
-        metricsContext.increment("kafka.bytes.read", message.message().payloadSize());
-        ILoggingEvent loggingEvent = serializer.fromBytes(message.message().payload());
-        // Use the message payload size as the size estimate of the logging event
-        // Although it's not the same as the in memory object size, it should be just a constant factor, hence
-        // it is proportional to the actual object size.
-        eventQueue.add(loggingEvent, loggingEvent.getTimeStamp(), message.message().payloadSize(), partition,
-                       new OffsetTime(message.nextOffset(), loggingEvent.getTimeStamp()));
-      } catch (IOException e) {
-        // This shouldn't happen. In case it happens (e.g. someone published some garbage), just skip the message.
-        LOG.trace("Fail to decode logging event from {}:{} at offset {}. Skipping it.",
-                  topic, partition, message.offset(), e);
-      }
-      processed = true;
-      offsets.put(partition, message.nextOffset());
-    }
-
-    return processed;
-  }
-
-  /**
-   * Fetches messages from Kafka across all partitions simultaneously.
-   */
-  private <T extends Map<Integer, Future<Iterable<MessageAndOffset>>>> T fetchAll(Int2LongMap offsets,
-                                                                                  T fetchFutures) {
-    for (final int partition : config.getPartitions()) {
-      final long offset = offsets.get(partition);
-
-      fetchFutures.put(partition, fetchExecutor.submit(() -> fetchMessages(partition, offset)));
-    }
-
-    return fetchFutures;
+    return process(topic, partition, messages);
   }
 
-  /**
-   * Appends buffered events to appender. If the {@code force} parameter is {@code false}, buffered events
-   * that are older than the buffer milliseconds will be appended and removed from the buffer.
-   * If {@code force} is {@code true}, then at least {@code maxQueueSize * MIN_FREE_FACTOR} events will be appended
-   * and removed, regardless of the event time.
-   *
-   * @return number of events appended to the appender
-   */
-  private int appendEvents(long currentTimeMillis, boolean forced) {
-    long minEventTime = currentTimeMillis - config.getEventDelayMillis();
-    long maxRetainSize = forced ? (long) (config.getMaxBufferSize() * MIN_FREE_FACTOR) : Long.MAX_VALUE;
-
-    TimeEventQueue.EventIterator<ILoggingEvent, OffsetTime> iterator = eventQueue.iterator();
-
-    int eventsAppended = 0;
-    long minDelay = Long.MAX_VALUE;
-    long maxDelay = -1;
+  private boolean process(String topic, int partition, Iterable<MessageAndOffset> messages) {
+    final byte[] processed = {0};
```

Comment:
Seems like you are using it as either 0 or 1. Better use a `boolean` or `AtomicBoolean`.